accord-core/src/main/java/accord/coordinate/CoordinationAdapter.java

```diff
@@ -261,21 +260,50 @@ public void execute(Node node, Topologies all, FullRoute<?> route, ExecutePath p
             }
         }
 
-        public static class InclusiveSyncPointAdapter<S extends Seekables<?, ?>> extends AbstractSyncPointAdapter<S>
+        private static class AbstractInclusiveSyncPointAdapter<S extends Seekables<?, ?>> extends AbstractSyncPointAdapter<S>
         {
-            private static final InclusiveSyncPointAdapter INSTANCE = new InclusiveSyncPointAdapter();
-        }
 
-        public static class InclusiveSyncPointBlockingAdapter<S extends Seekables<?, ?>> extends AbstractSyncPointAdapter<S>
-        {
-            private static final InclusiveSyncPointBlockingAdapter INSTANCE = new InclusiveSyncPointBlockingAdapter();
+            protected AbstractInclusiveSyncPointAdapter()
+            {
+                super();
+            }
 
             @Override
             public void execute(Node node, Topologies all, FullRoute<?> route, ExecutePath path, TxnId txnId, Txn txn, Timestamp executeAt, Deps deps, BiConsumer<? super SyncPoint<S>, Throwable> callback)
             {
                 ExecuteBlocking<S> execute = ExecuteBlocking.atQuorum(node, all, new SyncPoint<>(txnId, deps, (S)txn.keys(), route), executeAt);
-                execute.addCallback(callback);
                 execute.start();
+                // If there are no deps to wait on then everything is already applied
+                // If this is the async adapter then we want to invoke the callback immediately
+                // and the caller can wait on the txn locally if they want
+                if (deps.isEmpty() || this == AsyncInclusiveSyncPointAdapter.INSTANCE)
```

Comment:
I don't think `deps.isEmpty()` is a good test to apply as in a real cluster this should almost never be true; also not super keen on testing equality with an instance object, should maybe just have an overridable method?

Reply:
If you have a table that isn't written to frequently won't it only take minutes for everything to be truncated so there are empty reps? I was just thinking of making repair a little faster, but yeah I could see that the usage is infrequent enough we don't even really care.

I reduced constructor visibility to go with the `==` but yeah I'll make it a method.

Reply:
It's never empty, it will always have some exclusive sync point that represents the point in time of truncation to ensure no stale replica serves the request.

Perhaps we can issue entirely empty deps if we know no such replica exists, but I am not sure it is a sensible optimisation given the potential additional surface area for bugs.

Reply:
Updated it.